### ISSUE TITLE
Receiving release notifications for unavailable repositories.

### ISSUE THREAD

**Issue by tsvetiligo** [go-gitea/gitea#32677]
### Description

I still receive email notifications for repositories which I don't have access anymore (links in email show me 404).
I believe I don't suppose to get this notifications because release notes I see in email for private repositories may contain private information.

### Gitea Version

1.22.3

### How are you running Gitea?

docker

**Comment by yp05327** [comment]
Can you provide more details/information to help us reproduce this issue?
- the owner type of these repos (organization/individual)
-  > I don't have access anymore

For organization repos, removed from teams? For individual repos, removed from the contributor list?

**Comment by tsvetiligo** [comment]
The owner type of these repos in organization. I was removed from team.

**Comment by hiifong** [comment]
I used the main branch for testing, and I can continue to access the organization's private repository after being removed by the organization.
![image](https://github.com/user-attachments/assets/42e9c5d9-02b5-4df2-a755-fa8421e07c16)
![image](https://github.com/user-attachments/assets/11783a1c-cc26-48e6-a5fa-a05b8c61c5f0)

**Comment by hiifong** [comment]
> The owner type of these repos in organization. I was removed from team.

Could you provide detailed steps to reproduce?

**Comment by yp05327** [comment]
Related to the watch list? When user leave the org, but the watch list not updated, then he can also receive the notification mails.

**Comment by tsvetiligo** [comment]
> Related to the watch list? When user leave the org, but the watch list not updated, then he can also receive the notification mails.

Yes. Before I was removed from org, this repository was in my watch list. Now my watch list is empty.

**Comment by yp05327** [comment]
I check the code and tested in my local server. The watch list will be updated when user was remove from org. 🤔

**Comment by hiifong** [comment]
> I check the code and tested in my local server. The watch list will be updated when user was remove from org. 🤔 

If you are a super admin, watch list will not be updated.

**Comment by yp05327** [comment]
something is wrong in `func (env *accessibleReposEnv) RepoIDs(page, pageSize int) ([]int64, error)`, I will take a look.

**Comment by yp05327** [comment]
The root problem is that when we are using `ReconsiderWatches` to remove watches.
![image](https://github.com/user-attachments/assets/cdf798b2-4bc3-45d7-b01f-d28b2a1c05de)
It will check whether user can access these repos anymore. If not then remove the watch. But global admin has the top permission, so he can always access these repos. Then watch will not be removed.

**Comment by silverwind** [comment]
Release notifications now check current repository access, preventing mail to users who lost access after removal from an organization or team, even if a stale watch remains.

Fixed in https://github.com/go-gitea/gitea/commit/8a98ac22131cffef3d8ea23701167098da2772b4.

Written by Codex.
